test/dynamics/test_type_utils.py

```diff
@@ -256,22 +259,189 @@ def test_sparse_commutator_dissipator(self):
+        self.assertAllCloseSparse(to_csr(sparse_matrices), sparse_matrices)
+
+    def test_to_array_types(self):
+        """Type conversion tests for to_array"""
+        list_of_ops = [[[0, 1], [1, 0]], [[0, -1j], [1j, 0]], [[1, 0], [0, -1]]]
+        numpy_ops = np.array(list_of_ops)
+        normal_array = Array(np.array(list_of_ops))
+        op_arr = [Operator.from_label(s) for s in "XYZ"]
+        single_op = op_arr[0]
         list_of_arrays = [Array(op) for op in list_of_ops]
+        assert isinstance(to_array(numpy_ops), np.ndarray)
+        assert isinstance(to_array(normal_array), Array)
+        assert isinstance(to_array(op_arr), np.ndarray)
+        assert isinstance(to_array(single_op), np.ndarray)
+        assert isinstance(to_array(list_of_arrays), np.ndarray)
+
+    def test_to_csr_types(self):
+        """Type conversion tests for to_array"""
+        list_of_ops = [[[0, 1], [1, 0]], [[0, -1j], [1j, 0]], [[1, 0], [0, -1]]]
+        numpy_ops = np.array(list_of_ops)
+        normal_array = Array(np.array(list_of_ops))
         op_arr = [Operator.from_label(s) for s in "XYZ"]
+        single_op = op_arr[0]
+        list_of_arrays = [Array(op) for op in list_of_ops]
+        single_array = list_of_arrays[0]
         sparse_matrices = [csr_matrix(op) for op in list_of_ops]
-        self.assertAllClose(to_array(list_of_ops), normal_array)
-        self.assertAllClose(to_array(list_of_arrays), normal_array)
-        self.assertAllClose(to_array(op_arr), list_of_arrays)
+        assert isinstance(to_csr(normal_array)[0], csr_matrix)
+        assert isinstance(to_csr(numpy_ops)[0], csr_matrix)
+        assert isinstance(to_csr(op_arr)[0], csr_matrix)
+        assert isinstance(to_csr(single_op), csr_matrix)
+        assert isinstance(to_csr(single_array), csr_matrix)
+        assert isinstance(to_csr(list_of_arrays[0]), csr_matrix)
+        assert isinstance(to_csr(sparse_matrices), Iterable)
+        assert isinstance(to_csr(sparse_matrices)[0], csr_matrix)
+
+    def test_to_numeric_matrix_type(self):
+        """Tests for to_numeric_matrix_type"""
+        list_of_ops = [[[0, 1], [1, 0]], [[0, -1j], [1j, 0]], [[1, 0], [0, -1]]]
+        normal_array = Array(np.array(list_of_ops))
+        list_of_arrays = [Array(op) for op in list_of_ops]
+        op_arr = [Operator.from_label(s) for s in "XYZ"]
+        sparse_matrices = [csr_matrix(op) for op in list_of_ops]
+        self.assertAllClose(to_numeric_matrix_type(list_of_ops), normal_array)
+        self.assertAllClose(to_numeric_matrix_type(list_of_arrays), normal_array)
+        self.assertAllClose(to_numeric_matrix_type(op_arr), list_of_arrays)
         for i in range(3):
-            self.assertAllClose(sparse_matrices[i].toarray(), normal_array[i])
+            self.assertAllCloseSparse(
+                to_numeric_matrix_type(sparse_matrices)[i], sparse_matrices[i]
+            )
+
+
+class TestTypeUtilsQutip(TestTypeUtils, TestQutipBase):
```

Comment:
This test shouldn't inherit from `TestTypeUtils`, as it will rerun all of those tests, even though it's only adding a single test.

Set it to to just inherit from the usual dynamics base class.